client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -59,11 +64,18 @@ function PaymentMethodAdd(): ReactElement {
 	);
 
 	const returnQueryArg = useMemo(
-		() => getQueryArg( window.location.href, 'return' ),
+		() => ( getQueryArg( window.location.href, 'return' ) || '' ).toString(),
+		[ window.location.href, getQueryArg ]
+	);
+
+	const product = useMemo(
+		() => ( getQueryArg( window.location.href, 'product' ) || '' ).toString(),
 		[ window.location.href, getQueryArg ]
 	);
 
-	useReturnUrl( hasPaymentMethod );
+	const [ issueLicense, isSubmitting ] = useLicenseIssuing( null, product );
+
+	useReturnUrl( !! returnQueryArg && hasPaymentMethod );
```

Comment:
Only redirects in case there's a `return` query argument. So it will not redirect on the regular case, that there's only a `product` query argument or no argument at all. The `return` query argument will only be present when the user tries to assign a license without a payment method on file.